File: tutorials/testing/c++/linechart.cpp
#include "Bridges.h"
#include "LineChart.h"

using namespace bridges;

int main(int argc, char **argv) {

    Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

    LineChart plot;
	plot.setTitle("linechart plot");

	vector<double> x1{1, 3, 5, 20};
	vector<double> y1{2, 3, 5, 20};
	plot.setDataSeries("1", x1, y1);
	bridges.setDataStructure(plot);
	bridges.visualize();

	vector<double> x2{2., 15.2, 40.};
	vector<double> y2{4., 30.5, 400.99};
	plot.setDataSeries("2", x2, y2);
	bridges.visualize();


	plot.toggleLogarithmicX(true);

	bridges.visualize();

	plot.toggleMouseTrack(true);
	plot.toggleLogarithmicY(true);

	bridges.visualize();

	return 0;
}

File: tutorials/testing/python/linechart.py
from bridges.bridges import *
from bridges.line_chart import *
import sys
import array


def main():
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")


    # create the Bridges object, set credentials
    bridges.set_title ("Line chart test")

    plot = LineChart()
    plot.title = "Line Chart Plot"
    # set  the data
    x1 = [1, 3, 5, 20]
    y1 = [2, 3, 5, 20]
    plot.set_data_series("1", x1, y1)
    bridges.set_data_structure(plot)
    bridges.visualize()

    x2 = [2, 15.2,  40];
    y2 = [4, 30.5, 400.99];
    plot.set_data_series("2", x2, y2);
    bridges.visualize();

    plot.logarithmicx = True
    bridges.visualize();

    plot.mouse_track = True;
    plot.logarithmicy = True;

    bridges.visualize();


if __name__ == "__main__":
    main()
